cliquet/tests/test_storage.py

```diff
@@ -29,6 +29,17 @@ def __call__(self):
 
         self.assertRaises(AssertionError, Dumb)
 
+    def test_default_generator_allow_underscores_dash_alphabet(self):
+        class Dumb(generators.Generator):
+            def __call__(self):
+                return '1234'
```

Comment:
Not sure to understand why you actually need this. you probably can just test `generators.Generator().match()`